libpyvinyl/Parameters/Parameter.py

```diff
@@ -8,126 +8,168 @@ class Parameter(AbstractBaseClass):
         else:
-            print("WARNING: Value of parameter '" + self.name
-                  + "' illegal, ignored.")
+            raise ValueError("Value of parameter '" + self.name + "' illegal.")
 
-    def is_legal(self, value=None):
+    def is_legal(self, values=None):
         """
         Checks whether or not given or contained value is legal given constraints.
 
         Illegal intervals have the highest priority to be checked. Then it will check the
         legal intervals and options. The overlaps among the constrains will be overridden by
         the constrain of higher priority.
         """
-        if value is None:
-            value = self.value
-
-        # Check illegal intervals
-        for illegal_interval in self.illegal_intervals:
-            if illegal_interval[0] < value < illegal_interval[1]:
-                return False
-
-        # Check legal intervals
-        is_inside_a_legal_interval = False
-        for legal_interval in self.legal_intervals:
-            if legal_interval[0] < value < legal_interval[1]:
-                is_inside_a_legal_interval = True
 
-        if not is_inside_a_legal_interval and len(self.legal_intervals) > 0:
-            return False
+        truths = []
+        if values is None:
+            values = self._value
 
-        # checked intervals, can return if options not used (frequent case)
-        if len(self.options) == 0:
-            return True
+        if not hasattr(values, "__iter__") or isinstance(values, str):
+            #            print(str(hasattr(values, "__iter__")) + str(values))
+            value = values
 
-        for option in self.options:
-            if option == value:
-                # If the value matches any option, it is legal
-                return True
+            for option in self.options:
+                if option == value:
+                    #
+                    return self._options_are_legal
+
+            # Check illegal intervals
+            for interval in self.intervals:
+                if interval[0] < value < interval[1]:
+                    #
+                    return self._intervals_are_legal
+            # T: are legal values
+            # F: are illegal values
+            # 0: empty list
+            #
+            # P I -> truth
+            # T T -> F
+            # F T -> F
+            # 0 T -> F
+            # T F -> T
+            # F F -> T
+            # 0 F -> T
+            # T 0 -> F
+            # F 0 -> T
+            # 0 0 -> T
+
+            return (not self._intervals_are_legal and len(self.intervals) > 0) or (
+                len(self.intervals) == 0
+                and (not self._options_are_legal or len(self.options) == 0)
+            )
+
+        # else
+        for value in values:
+            truths.append(self.is_legal(value))
+
+        # all values have to be True
+        for truth in truths:
+            if truth is False:
+                return False
```

Comment:
I don't think there is any need for a truth list, we can stop checking values as soon as one illegal value is found without a need to evaluate the remaining.

```suggestion
for value in values:
    if not self.is_legal(value):
        return False

return True
```

Reply:
Yes.